base/est/src/main/java/org/dogtagpki/est/ExternalProcessRequestAuthorizer.java

```diff
@@ -70,17 +76,27 @@ public void start() throws Throwable {
     public Object authorizeSimpleenroll(
         ESTRequestAuthorizationData data, PKCS10 csr)
             throws PKIException {
+        if (enrollMatchSubjSAN && data.clientCertChain != null && data.clientCertChain.length > 0) {
+            ensureCSRMatchesToBeCert(csr, data.clientCertChain[0], false);
+        }
+        if (enrollMatchSubjSAN && data.clientCertChain == null) {
+            if(data.principal instanceof PKIPrincipal principal) {
+                ensureCSRMatchesToBeCert(csr, principal.getUser());
+            }
+        }
```

Comment:
Just to make sure we cover all cases (since this is an authorization code), could we change the code into this?
```
if (enrollMatchSubjSAN) {
    if (data.clientCertChain != null && data.clientCertChain.length > 0) {
        ensureCSRMatchesToBeCert(csr, data.clientCertChain[0], false);
    } else {
        PKIPrincipal principal = (PKIPrincipal) data.principal;
        ensureCSRMatchesToBeCert(csr, principal.getUser());
    }
}
```